config/config.go

```diff
@@ -1,9 +1,105 @@
 package config
 
-import "github.com/klarna/eremetic/database"
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
 
+	"github.com/kardianos/osext"
+	"github.com/kelseyhightower/envconfig"
+	"gopkg.in/yaml.v2"
+
+	"github.com/klarna/eremetic/database"
+)
+
+// The Config struct holds the Eremetic Configuration
 type Config struct {
-	Version   string
-	BuildDate string
-	Database  database.TaskDB
+	// General
+	Version   string `yaml:"-"`
+	BuildDate string `yaml:"-"`
+
+	// Logging
+	LogLevel  string `yaml:"loglevel"`
+	LogFormat string `yaml:"logformat"`
+
+	// Server
+	Address string `yaml:"address"`
+	Port    int    `yaml:"port"`
+
+	// Database
+	Database       database.TaskDB `yaml:"-" ignored:"true"`
```

Comment:
Sure thing but could you at least keep them as distinct types. Config/OtherConfig?